packages/react-components/react-utilities/src/utils/types.ts

```diff
@@ -134,3 +134,45 @@ export type PropsWithoutChildren<P> = 'children' extends keyof P ? DistributiveO
  * **THIS TYPE IS INTERNAL AND SHOULD NEVER BE EXPOSED**
  */
 export type ReactVersionDependent<Modern, Legacy> = {} extends React.ReactNode ? Legacy : Modern;
+
+/**
+ * Our own alias for `JSX.Element` type that is compatible with both React 17 and React >=18.
+ * Use this type when annotating JSX markup in all our code in order to avoid issues between different React versions.
+ *
+ * Example usage:
+ *
+ * BAD:
+ * ```tsx
+ * const renderFoo = (state: FooState) = <div {...props}>Hello World</div>;
+ * // infers
+ * // R17:  declare const renderFoo: (state: FooState) => JSX.Element;
+ * // R18+: declare const renderFoo: (state: FooState) => React.JSX.Element;
+ * ```
+ *
+ * GOOD:
+ * ```tsx
+ * import type { JSXElement } from '@fluentui/react-utilities';
+ * const renderFoo = (state: FooState): JSXElement = <div {...props}>Hello World</div>;
+ * ```
+ */
+export type JSXElement = React.ReactElement<
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  any,
+  any
+  /* eslint-enable @typescript-eslint/no-explicit-any */
+>;
+
+/**
+ * Our own alias for `JSX.IntrinsicElements` type that is compatible with both React 17 and React >=18.
+ * Use this type to get the intrinsic elements from React types in order to avoid issues between different React versions.
+ */
+export type JSXIntrinsicElements =
```

Comment:
NOTE: this will not work as expected because `tsc` will strip these `@ts-ignores` on emit.

options to solve this:
- create a postBuild hook within `generate-api` executor to "search and replace" parts of emitted `.d.ts`
- move the definition to separate file and override emitted `.d.ts` with our own version that will contain   `@ts-ignore` pragmas

Reply:
final approach: leveraging TSC type checker to generated unwrapped union